src/platform/plugins/shared/inspector/common/adapters/request/request_adapter.ts

```diff
@@ -21,10 +21,12 @@ import { Request, RequestParams, RequestStatus } from './types';
  */
 export class RequestAdapter extends EventEmitter {
   private requests: Map<string, Request>;
+  private responses: WeakMap<Request, RequestResponder>;
 
   constructor() {
     super();
     this.requests = new Map();
+    this.responses = new WeakMap();
```

Comment:
Why is responses a WeakMap instead of just a map keyed to the request id?

Reply:
I thought about that, but:
* there's no need to list the entries
* using the `request` itself saves a `map` cycle altogether
* it could `reset` as side effect automatically

Given the use cases it makes no specific difference between the two, other than no providing the iterator API.